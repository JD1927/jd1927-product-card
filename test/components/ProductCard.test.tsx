import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product1 } from '../data/products';
const { act, create } = renderer;

describe('ProductCard', () => {
  test('Should show the component successfully', () => {
    // Arrange
    const wrapper = create(
      <ProductCard product={product1}>
        {
          () => (<h1>Product Card JD1927</h1>)
        }
      </ProductCard>
    );
    // Act
    // Assert
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('Should increase the counter', () => {
    // Arrange
    const wrapper = create(
      <ProductCard
        product={product1}
        initialValues={{ count: 4, maxCount: 10 }}>
        {
          ({ count, increaseBy, maxCount }) => (
            <>
              <h1>Product Card JD1927</h1>
              <span>{count}</span>
              <span>{maxCount}</span>
              <button onClick={() => increaseBy(1)}></button>
            </>
          )
        }
      </ProductCard>
    );
    // Act
    let tree = wrapper.toJSON() as any;
    // Assert
    expect(tree).toMatchSnapshot();

    act(() => {
      tree.children[3].props.onClick();
    });
    tree = wrapper.toJSON();

    expect(tree.children[2].children[0]).toBe('10');
    expect(tree.children[1].children[0]).toBe('4');
  });
});
