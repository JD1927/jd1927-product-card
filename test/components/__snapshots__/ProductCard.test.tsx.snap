// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Should increase the counter 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card JD1927
  </h1>
  <span>
    4
  </span>
  <span>
    10
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`ProductCard Should show the component successfully 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card JD1927
  </h1>
</div>
`;
